jax/interpreters/mlir.py

```diff
@@ -977,13 +977,13 @@ def jaxpr_subcomp(ctx: ModuleContext, jaxpr: core.Jaxpr,
   Assumes that an MLIR context, location, and insertion point are set.
   """
   assert ctx.platform != "gpu"
-  def read(v: core.Var) -> Sequence[ir.Value]:
+  def read(v: Union[core.Var, core.Literal]) -> Sequence[ir.Value]:
     if type(v) is core.Literal:
       return ir_constants(v.val, canonicalize_types=True)
     else:
-      return env[v]
+      return env[cast(core.Var, v)]
```

Comment:
Can we write an `assert isinstance(v, core.Var)` instead of `cast`? I hate `cast` :P

Reply:
Done